File: IT/Input-Output/Esercizi/Java/Esercizio26.java
/* file Esercizio26.java */

import java.io.Console;
import java.io.File;
import java.io.FileReader;
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.IOException;

public class Esercizio26 {
	public static void main(String [] args) {
		Console console = System.console();
		String filename = console.readLine("Apri il file: ");

		try {
			File doc = new File(filename);

			FileReader reader = new FileReader(doc);
			BufferedReader buffer = new BufferedReader(reader);

			String s = console.readLine("Parola da cercare: ");

			boolean found = false;
			int i = 1;

			while (buffer.ready()) {
				String line = buffer.readLine();
				if (line.toLowerCase().indexOf(s.toLowerCase()) != -1) {
					if (!found) found = true;
					System.out.println("La parola \"" + s + "\" compare alla riga " + i + ": \"" + line + "\"");
				}
				i++;
			}

			if (!found) {
				System.out.println("Parola \"" + s + "\" non trovata!");
			}

			buffer.close();
		}
		catch (FileNotFoundException e) {
			System.out.println(e);
		}
		catch (IOException e) {
			System.out.println(e);
		}
	}
}

File: IT/Input-Output/Esercizi/C++/esercizio26.cpp
/*
file esercizio26.cpp
compilare con: g++ -std=c++11 esercizio26.cpp
*/

#include <iostream>
#include <fstream>
#include <algorithm>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::ifstream;
using std::transform;

int main()
{
	cout << "Apri il file: ";

	string filename;
	getline(cin, filename);

	ifstream f(filename);

	if (f) {
		cout << "Parola da cercare: ";

		string s;
		getline(cin, s);

		string copy = s;
		transform(copy.begin(), copy.end(), copy.begin(), ::tolower);

		bool found = false;
		int i = 1;

		string line;
		while (getline(f, line)) {
			string lcopy = line;
			transform(lcopy.begin(), lcopy.end(), lcopy.begin(), ::tolower);
			if (lcopy.find(copy) != -1) {
				if (!found) found = true;
				cout << "La parola \"" << s << "\" compare alla riga " << i << ": \"" << line << "\"" << endl;
			}
			i++;
		}

		if (!found) {
			cout << "Parola \"" << s << "\" non trovata!" << endl;
		}
	}
	else {
		cout << "Impossibile aprire il file!" << endl;
	}
}

